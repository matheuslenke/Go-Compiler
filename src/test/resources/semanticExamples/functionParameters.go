package samples

import "fmt"

func ParamTest(a, b int) {
	// var d int        // +
	// var n, c float64 // + strange extra levels
	// // var d = 1        // + doesn't show zero value
	// var e, f float32 = -1.0, -2.5 // +
	// var numero int = 300
	// var pedroPassou bool = true
	a = 5
}

func ParamTest2(param1,param2 string, param3 float32) {
	fmt.Println(param3);
}

func ParamTest3(param int) (result1, result2 float32) {
	var returnVar float32 = 1.0
	return returnVar
	result1 = 2.5
}

func ParamTest4() (string, string) {
	return "Testando", "Testando"
}