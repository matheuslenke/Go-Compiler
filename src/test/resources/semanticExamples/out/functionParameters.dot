digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) ParamTest@5"];
node11[label="parameters"];
node12[label="(int) a@6"];
node11 -> node12;
node13[label="(int) b@7"];
node11 -> node13;
node10 -> node11;
node14[label="block"];
node15[label="assign_list_node"];
node16[label="="];
node17[label="(int) a@6"];
node16 -> node17;
node18[label="(int) int@5"];
node16 -> node18;
node15 -> node16;
node14 -> node15;
node10 -> node14;
node0 -> node10;
node19[label="(func) ParamTest2@8"];
node20[label="parameters"];
node21[label="(string) param1@9"];
node20 -> node21;
node22[label="(string) param2@10"];
node20 -> node22;
node23[label="(float) param3@11"];
node20 -> node23;
node19 -> node20;
node24[label="block"];
node25[label="write"];
node26[label="Println@1"];
node25 -> node26;
node27[label="arguments"];
node28[label="(string) string@0"];
node27 -> node28;
node25 -> node27;
node24 -> node25;
node19 -> node24;
node0 -> node19;
node29[label="(func) ParamTest3@12"];
node30[label="parameters"];
node31[label="(int) param@13"];
node30 -> node31;
node29 -> node30;
node32[label="result"];
node33[label="(float) type_use"];
node32 -> node33;
node29 -> node32;
node34[label="block"];
node35[label="var_list"];
node36[label="="];
node37[label="(float) returnVar@14"];
node36 -> node37;
node38[label="(float) real@1.0"];
node36 -> node38;
node35 -> node36;
node34 -> node35;
node39[label="return"];
node40[label="(float) returnVar@14"];
node39 -> node40;
node34 -> node39;
node29 -> node34;
node0 -> node29;
node41[label="(func) ParamTest4@15"];
node42[label="parameters"];
node41 -> node42;
node43[label="result"];
node44[label="(string) type_use"];
node43 -> node44;
node41 -> node43;
node45[label="block"];
node46[label="return"];
node47[label="(string) string@1"];
node46 -> node47;
node45 -> node46;
node41 -> node45;
node0 -> node41;
}
