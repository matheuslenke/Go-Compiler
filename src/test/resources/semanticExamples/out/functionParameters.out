[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: String teste
Entry 1: Testando




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: ParamTest, line: 5, type: func, scopeName: global
Entry 6 -- name: a, line: 5, type: int, scopeName: ParamTest
Entry 7 -- name: b, line: 5, type: int, scopeName: ParamTest
Entry 8 -- name: ParamTest2, line: 15, type: func, scopeName: global
Entry 9 -- name: param1, line: 15, type: string, scopeName: ParamTest2
Entry 10 -- name: param2, line: 15, type: string, scopeName: ParamTest2
Entry 11 -- name: param3, line: 15, type: float, scopeName: ParamTest2
Entry 12 -- name: ParamTest3, line: 19, type: func, scopeName: global
Entry 13 -- name: param, line: 19, type: int, scopeName: ParamTest3
Entry 14 -- name: returnVar, line: 20, type: float, scopeName: ParamTest3
Entry 15 -- name: ParamTest4, line: 24, type: func, scopeName: global




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.802 s
[INFO] Finished at: 2022-02-24T20:53:50-03:00
[INFO] ------------------------------------------------------------------------
